### PR TITLE
Migrate SchedulerSource to use generated reconciler.

### PR DESCRIPTION
**Pull request by nlopezgi**
3rd PR part of # https://github.com/google/knative-gcp/issues/557

## Proposed Changes

- 🎁 Use reconciler gen to implement strongly typed reconcilers.
- Also remove finalizerName from pubsub reconciler as

### CODE REVIEW

File: pkg/reconciler/events/scheduler/scheduler.go [hunk 1]
```diff
@@ -63,117 +58,29 @@ type Reconciler struct {
...
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledPubSubFailedReason, "Reconcile PubSub failed with: %s", err)
 	}
 
 	jobName := resources.GenerateJobName(scheduler)
 	err = r.reconcileJob(ctx, scheduler, topic, jobName)
 	if err != nil {
-		scheduler.Status.MarkJobNotReady("JobReconcileFailed", "Failed to reconcile CloudSchedulerSource job: %s", err.Error())
-		return err
+		scheduler.Status.MarkJobNotReady(reconciledFailedReason, "Failed to reconcile CloudSchedulerSource job: %s", err.Error())
+		return reconciler.NewEvent(corev1.EventTypeWarning, reconciledFailedReason, "Reconcile Job failed with: %s", err)
```
Comment (nachocano) [inline pkg/reconciler/events/scheduler/scheduler.go]: err.Error()

File: pkg/reconciler/events/scheduler/scheduler.go [hunk 2]
```diff
@@ -63,117 +58,29 @@ type Reconciler struct {
...
 	}
 	scheduler.Status.MarkJobReady(jobName)
-	return nil
+	return reconciler.NewEvent(corev1.EventTypeNormal, reconciledSuccessReason, "CloudSchedulerSource reconciled: %s", scheduler.Status.JobName)
```
Comment (nachocano) [inline pkg/reconciler/events/scheduler/scheduler.go]: `CloudSchedulerSource reconciled: "%s/%s"`, scheduler.Namespace, scheduler.Name)

File: pkg/reconciler/events/scheduler/scheduler.go (hunk 1 above)
Comment (nachocano) [reply pkg/reconciler/events/scheduler/scheduler.go]: here and other places where we are passing just err with %s format...

File: pkg/reconciler/events/scheduler/scheduler.go [hunk 3]
```diff
@@ -265,96 +172,15 @@ func (r *Reconciler) deleteJob(ctx context.Context, scheduler *v1alpha1.CloudSch
...
-// updateFinalizers is a generic method for future compatibility with a
-// reconciler SDK.
-func (r *Reconciler) updateFinalizers(ctx context.Context, desired *v1alpha1.CloudSchedulerSource) (*v1alpha1.CloudSchedulerSource, bool, error) {
-	scheduler, err := r.schedulerLister.CloudSchedulerSources(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, false, err
+func (r *Reconciler) FinalizeKind(ctx context.Context, scheduler *v1alpha1.CloudSchedulerSource) reconciler.Event {
+	logging.FromContext(ctx).Desugar().Debug("Deleting CloudSchedulerSource job")
+	if err := r.deleteJob(ctx, scheduler); err != nil {
+		return reconciler.NewEvent(corev1.EventTypeWarning, deleteJobFailed, "Failed to delete CloudSchedulerSource job: %s", err)
```
Comment (nachocano) [inline pkg/reconciler/events/scheduler/scheduler.go]: nit: err.Error()

File: pkg/reconciler/events/scheduler/scheduler.go [hunk 4]
```diff
@@ -265,96 +172,15 @@ func (r *Reconciler) deleteJob(ctx context.Context, scheduler *v1alpha1.CloudSch
...
-		finalizers = existingFinalizers.List()
-	}
-
-	mergePatch := map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"finalizers":      finalizers,
-			"resourceVersion": existing.ResourceVersion,
-		},
+	if err := r.PubSubBase.DeletePubSub(ctx, scheduler); err != nil {
+		return reconciler.NewEvent(corev1.EventTypeWarning, deletePubSubFailed, "Failed to delete CloudStorageSource PubSub: %s", err)
```
Comment (nachocano) [inline pkg/reconciler/events/scheduler/scheduler.go]: nit: err.Error()

**Comment by nachocano** [comment]
/test pull-google-knative-gcp-integration-tests

/approve
/lgtm

/hold in case you want to address the cosmetic comments now, otherwise unhold, and do in the following one... not a big deal
Thanks for adding this @nlopezgi

File: pkg/reconciler/events/scheduler/scheduler.go (hunk 1 above)
Comment (nlopezgi) [reply pkg/reconciler/events/scheduler/scheduler.go]: done

File: pkg/reconciler/events/scheduler/scheduler.go (hunk 2 above)
Comment (nlopezgi) [reply pkg/reconciler/events/scheduler/scheduler.go]: done

File: pkg/reconciler/events/scheduler/scheduler.go (hunk 3 above)
Comment (nlopezgi) [reply pkg/reconciler/events/scheduler/scheduler.go]: done

File: pkg/reconciler/events/scheduler/scheduler.go (hunk 4 above)
Comment (nlopezgi) [reply pkg/reconciler/events/scheduler/scheduler.go]: done

File: pkg/reconciler/events/scheduler/scheduler.go [hunk 5]
```diff
@@ -175,11 +175,11 @@ func (r *Reconciler) deleteJob(ctx context.Context, scheduler *v1alpha1.CloudSch
 func (r *Reconciler) FinalizeKind(ctx context.Context, scheduler *v1alpha1.CloudSchedulerSource) reconciler.Event {
 	logging.FromContext(ctx).Desugar().Debug("Deleting CloudSchedulerSource job")
 	if err := r.deleteJob(ctx, scheduler); err != nil {
-		return reconciler.NewEvent(corev1.EventTypeWarning, deleteJobFailed, "Failed to delete CloudSchedulerSource job: %s", err)
+		return reconciler.NewEvent(corev1.EventTypeWarning, deleteJobFailed, "Failed to delete CloudSchedulerSource job: %s", err.Error())
 	}
 
 	if err := r.PubSubBase.DeletePubSub(ctx, scheduler); err != nil {
-		return reconciler.NewEvent(corev1.EventTypeWarning, deletePubSubFailed, "Failed to delete CloudStorageSource PubSub: %s", err)
+		return reconciler.NewEvent(corev1.EventTypeWarning, deletePubSubFailed, "Failed to delete CloudStorageSource PubSub: %s", err.Error())
```
Comment (nachocano) [inline pkg/reconciler/events/scheduler/scheduler.go]: nit: CloudStorageSource -> CloudSchedulerSource

Comment (nlopezgi) [reply pkg/reconciler/events/scheduler/scheduler.go]: fixed

**Comment by nachocano** [comment]
/lgtm

Thanks @nlopezgi !!!

**Pull request merged**